network-api/networkapi/wagtailpages/utils.py

```diff
@@ -114,39 +117,44 @@ def get_content_related_by_tag(page, result_count=3):
     on its `.tags` content. If it has tags.
     """
 
-    tagged_post_types = (
+    if hasattr(page.specific, 'tags') is False:
+        return list()
+
+    page_models_with_tags = (
         apps.get_model('wagtailpages', 'BlogPage'),
-        # Add more models here as we add tags to more types
+        apps.get_model('wagtailpages', 'BanneredCampaignPage'),
     )
 
+    results = False
     own_tags = page.tags.all()
 
-    related_qs = False
+    for page_type in page_models_with_tags:
+        # get all pages that share tags with this page
+        related_pages = page_type.objects.filter(tags__in=own_tags)
 
-    for post_type in tagged_post_types:
-        related_posts = post_type.objects.filter(tags__in=own_tags).distinct()
+        # Exlude "this page" from the result set for the page's own page type
+        # so that we don't end up with "this page is most similar to itself".
+        if page.__class__ is page_type:
+            related_pages = related_pages.exclude(pk=page.pk)
 
-        # Exlude "this page" from the result set for the page's own.
-        # page type, so we don't yield "this page is similar to itself".
-        if page.__class__ is post_type:
-            related_posts = related_posts.exclude(pk=page.pk)
+        # Annotate the results by adding a column that, effectively, records
+        # a `countDistinct(result_table.tag)` for each related post.
```

Comment:
(just because I don't know Django that well) What is `countDistinct(result_table.tag)`? Does it give you the number of distinct common tags that a related post shared with the current page?

Reply:
yep, that's exactly what it does.